stac_model/input.py

```diff
@@ -18,27 +18,58 @@ def validate_dimensions(self) -> Self:
         return self
 
 
-class MLMStatistic(MLMBaseModel):  # FIXME: add 'Statistics' dep from raster extension (cases required to be triggered)
-    minimum: Annotated[Optional[Number], OmitIfNone] = None
-    maximum: Annotated[Optional[Number], OmitIfNone] = None
-    mean: Annotated[Optional[Number], OmitIfNone] = None
-    stddev: Annotated[Optional[Number], OmitIfNone] = None
-    count: Annotated[Optional[int], OmitIfNone] = None
-    valid_percent: Annotated[Optional[Number], OmitIfNone] = None
+class ScalingClipMin(MLMBaseModel):
+    type: Literal["clip-min"] = "clip-min"
+    minimum: Number
 
 
-NormalizeType: TypeAlias = Optional[
-    Literal[
-        "min-max",
-        "z-score",
-        "l1",
-        "l2",
-        "l2sqr",
-        "hamming",
-        "hamming2",
-        "type-mask",
-        "relative",
-        "inf",
+class ScalingClipMax(MLMBaseModel):
+    type: Literal["clip-max"] = "clip-max"
+    maximum: Number
+
+
+class ScalingClip(MLMBaseModel):
+    type: Literal["clip"] = "clip"
+    minimum: Number
+    maximum: Number
+
+
+class ScalingMinMax(MLMBaseModel):
+    type: Literal["min-max"] = "min-max"
+    minimum: Number
+    maximum: Number
+
+
+class ScalingZScore(MLMBaseModel):
+    type: Literal["z-score"] = "z-score"
+    mean: Number
+    stddev: Number
+
+
+class ScalingOffset(MLMBaseModel):
+    type: Literal["offset"] = "offset"
+    value: Number
+
+
+class ScalingScale(MLMBaseModel):
+    type: Literal["scale"] = "scale"
+    value: Number
+
+
+class ScalingProcessingExpression(ProcessingExpression):
+    type: Literal["processing"] = "processing"
+
+
+ScalingObject: TypeAlias = Optional[
+    Union[
+        ScalingMinMax,
+        ScalingZScore,
+        ScalingClip,
+        ScalingClipMin,
+        ScalingClipMax,
+        ScalingOffset,
+        ScalingScale,
+        ScalingProcessingExpression,
```

Comment:
Is it confusing for users to be able to provide processing expressions in two different locations within the ModelInputObject? I thought that processing expressions would still only be supplied via the existing pre_processing_function field?

Reply:
I'm guessing using processing expressions is for advanced use cases anyway, so I would expect only more advanced users to define them anyway. It at least offers some flexibility for overall-resize/scaling vs per-band-resize/scaling expressions. Before, per-band expression would have needed to be embedded within the overall expression, which made it itself way more convoluted.